### ISSUE TITLE
question about train/test target

### ISSUE THREAD

**Issue by wangqiim** [LongguangWang/SMSR#20]
thanks for your work, it is interesting! I have some question about train & test(figure 5 in paper). 
1) In paper, the figure.5 looks like a (0 or 1) mask in training. But I find the code (mask module) it just like a attention module(pixel attention and channel attention)
2) Because question 1), the test target and train target are different. In training, the pixel and channel just times a weight (not zero through mask module), But in test, it skip the computation(times zero or one).
3) The channel mask will change through input sample (in paper) , But the code looks like it is a invariable after trained. So it is just a channel prune?https://github.com/The-Learning-And-Vision-Atelier-LAVA/SMSR/blob/daac49c9a107778c95e11a16fd5b4a8b45513678/model/smsr.py#L58

**Comment by LongguangWang** [comment]
Hi @wangqiim, thanks for your interest in our work.
1. During training, we use a gating mechanism to mimick sparse convolution to maintain the gradients at all locations. As Gumbel softmax distribution approaches one-hot distribution, the masks gradually become binary.
2. Following the above,  the binary masks used during training and sparse convolutions used during test produce consistent results.
3. A static channel mask is used in our method and please refer to the supplemental material for detailed analyses.

**Comment by wangqiim** [comment]
> Hi @wangqiim, thanks for your interest in our work.
> 
> 1. During training, we use a gating mechanism to mimick sparse convolution to maintain the gradients at all locations. As Gumbel softmax distribution approaches one-hot distribution, the masks gradually become binary.
> 2. Following the above,  the binary masks used during training and sparse convolutions used during test produce consistent results.
> 3. A static channel mask is used in our method and please refer to the supplemental material for detailed analyses.

Make sense! Thank you!
